src/test/utils/BaseTest.sol

```diff
@@ -202,4 +202,65 @@ abstract contract BaseTest is DSTest, stdCheats {
     function getWallet() public returns (Wallet wallet) {
         wallet = new Wallet();
     }
+
+    function assertBal721(
```

Comment:
i wonder if we should generalized this to
`assertBalanceEq(address _token, address _owner, uint256[] memory _tokenIds, uint256[] memory _amounts);`
`assertBalanceGt(address _token, address _owner, uint256[] memory _tokenIds, uint256[] memory _amounts);`
`assertBalanceLt(address _token, address _owner, uint256[] memory _tokenIds, uint256[] memory _amounts);`

it introspect (erc165) to check which token it is and compare the balance.

Reply:
for erc721, if amounts = 1, check owner of.
the helper function check for balance smartly.

Reply:
I don't like this because the function signature does not conceptually make sense for every ERCxxx e.g. the concept of `amounts` doesn't make sense for ERC721, and the concept of `tokenIds` doesn't make sense for ERC20.  I think this in turn harms the readability of tests.

Reply:
what do you think about opting to be `assertBalanceERC721` then? 
instead of 
```
if (_assertIsOwner) {
                assertTrue(isOwnerOfToken);
            } else {
                assertTrue(!isOwnerOfToken);
            }
            ```
            
            we should make turn them into two functions:
            `assertERC721IsOwner(...)` `assertERC721IsNotOwner(...)`

Reply:
rules of thumb, an assertion should be only assert 1 rule. it should not take in params and flip the assertion rule.